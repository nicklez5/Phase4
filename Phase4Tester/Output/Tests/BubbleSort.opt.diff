--- SelfTestCases/BubbleSort.opt.out	2011-04-28 14:55:21.000000000 -0700
+++ ./Output/Tests/BubbleSort.opt.out	2020-03-14 13:04:23.501787434 -0700
@@ -1,23 +1,12 @@
-20
-7
-12
-18
-2
-11
-6
-9
-19
-5
-99999
-2
-5
-6
-7
-9
-11
-12
-18
-19
-20
-0
+Error in /home/jackson/Desktop/phase4/Phase4Tester/./Output/Tests/BubbleSort.opt.compiler.out line 244 column 11: "0": operand is of incorrect type
+Error in /home/jackson/Desktop/phase4/Phase4Tester/./Output/Tests/BubbleSort.opt.compiler.out line 258 column 11: "1": operand is of incorrect type
+Error in /home/jackson/Desktop/phase4/Phase4Tester/./Output/Tests/BubbleSort.opt.compiler.out line 272 column 11: "2": operand is of incorrect type
+Error in /home/jackson/Desktop/phase4/Phase4Tester/./Output/Tests/BubbleSort.opt.compiler.out line 286 column 11: "3": operand is of incorrect type
+Error in /home/jackson/Desktop/phase4/Phase4Tester/./Output/Tests/BubbleSort.opt.compiler.out line 300 column 11: "4": operand is of incorrect type
+Error in /home/jackson/Desktop/phase4/Phase4Tester/./Output/Tests/BubbleSort.opt.compiler.out line 314 column 11: "5": operand is of incorrect type
+Error in /home/jackson/Desktop/phase4/Phase4Tester/./Output/Tests/BubbleSort.opt.compiler.out line 328 column 11: "6": operand is of incorrect type
+Error in /home/jackson/Desktop/phase4/Phase4Tester/./Output/Tests/BubbleSort.opt.compiler.out line 342 column 11: "7": operand is of incorrect type
+Error in /home/jackson/Desktop/phase4/Phase4Tester/./Output/Tests/BubbleSort.opt.compiler.out line 356 column 11: "8": operand is of incorrect type
+Error in /home/jackson/Desktop/phase4/Phase4Tester/./Output/Tests/BubbleSort.opt.compiler.out line 370 column 11: "9": operand is of incorrect type
 
+Processing terminated due to errors.
