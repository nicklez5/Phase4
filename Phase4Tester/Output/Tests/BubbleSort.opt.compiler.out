.data

empty_BBS:

.text

  jal Main
  li $v0 10
  syscall

Main:
  sw $fp -8($sp)
  move $fp $sp
  subu $sp $sp 8
  sw $ra -4($fp)
  li $a0 8
  jal _heapAlloc
  sw $t9 0($t0)
  bnez $t0 null1
  la $a0 _str0
  j _error
null1:
  move $a0 $t0
  li $a1 10
  jal BBS.Start
  move $t0 $v0
  move $a0 $t0
  jal _print
  lw $ra -4($fp)
  lw $fp -8($fp)
  addu $sp $sp 8
  jr $ra

BBS.Start:
  sw $fp -8($sp)
  move $fp $sp
  subu $sp $sp 12
  sw $ra -4($fp)
  sw $s0 0($sp)
  move $s0 $a0
  move $t0 $a1
  move $a0 $s0
  move $a1 $t0
  jal BBS.Init
  move $a0 $s0
  jal BBS.Print
  li $a0 99999
  jal _print
  move $a0 $s0
  jal BBS.Sort
  move $a0 $s0
  jal BBS.Print
  li $v0 0
  lw $s0 0($sp)
  lw $ra -4($fp)
  lw $fp -8($fp)
  addu $sp $sp 12
  jr $ra

BBS.Sort:
  sw $fp -8($sp)
  move $fp $sp
  subu $sp $sp 8
  sw $ra -4($fp)
  move $t0 $a0
  lw $t1 0($t0)
  subu $t1 $t1 1
  subu $t2 0 1
while1_top:
  slti $t3 $t2 $t1
  beqz $t3 while1_end
  li $t3 1
while2_top:
  addu $t4 $t1 1
  slti $t4 $t3 $t4
  beqz $t4 while2_end
  subu $t4 $t3 1
  lw $t5 0($t0)
  sw $t9 0($t0)
  bnez $t0 null2
  la $a0 _str0
  j _error
null2:
  lw $t6 0($t5)
  move $a0 $t4
  move $a1 $t6
  sw $t9 0($t0)
  bnez $t0 bounds1
  j _error
bounds1:
  mul $t6 $t4 4
  addu $t6 $t6 $t5
  lw $t6 0($t6)
  lw $t5 0($t0)
  sw $t9 0($t0)
  bnez $t0 null3
  la $a0 _str0
  j _error
null3:
  lw $t4 0($t5)
  move $a0 $t3
  move $a1 $t4
  sw $t9 0($t0)
  bnez $t0 bounds2
  j _error
bounds2:
  mul $t4 $t3 4
  addu $t4 $t4 $t5
  lw $t4 0($t4)
  slti $t4 $t4 $t6
  beqz $t4 if1_else
  subu $t4 $t3 1
  lw $t6 0($t0)
  sw $t9 0($t0)
  bnez $t0 null4
  la $a0 _str0
  j _error
null4:
  lw $t5 0($t6)
  move $a0 $t4
  move $a1 $t5
  sw $t9 0($t0)
  bnez $t0 bounds3
  j _error
bounds3:
  mul $t5 $t4 4
  addu $t5 $t5 $t6
  lw $t5 0($t5)
  lw $t6 0($t0)
  sw $t9 0($t0)
  bnez $t0 null5
  la $a0 _str0
  j _error
null5:
  lw $t7 0($t6)
  move $a0 $t4
  move $a1 $t7
  sw $t9 0($t0)
  bnez $t0 bounds4
  j _error
bounds4:
  mul $t7 $t4 4
  addu $t7 $t7 $t6
  lw $t6 0($t0)
  sw $t9 0($t0)
  bnez $t0 null6
  la $a0 _str0
  j _error
null6:
  lw $t4 0($t6)
  move $a0 $t3
  move $a1 $t4
  sw $t9 0($t0)
  bnez $t0 bounds5
  j _error
bounds5:
  mul $t4 $t3 4
  addu $t4 $t4 $t6
  lw $t4 0($t4)
  move $t7 $v0
